rfcs/convergence/native-element-props.md

```diff
@@ -0,0 +1,310 @@
+# RFC: Customizing application of native element props and `ref`
+
+@ecraig12345 @sopranopillow @behowell
+
+## Summary
+
+Instead of always applying native props and `ref` to the root, allow setting a different "primary" slot where these props should be applied. This option should be used sparingly: for input components and possibly a few other cases (options for exact criteria discussed below).
+
+Also explicitly expose the `root` slot for all components. This will allow passing native props to the root if needed for the "special case" components discussed above, and provide consistency across the library.
```

Comment:
Do I correctly understand that this RFC is proposing to do following?

```tsx
<>
  <Button root={{ type: "submit" }} />
  <Link root={{ href: "#" }} />
  <Accordion root={{ onClick: () => {} }} />
</>
```

If so, I don't think it's a good idea as it will create confusion for customers. For example:

```tsx
<>
  {/* 😕 will this render <a href="#bar" />? */}
  <Link href="#bar" root={{ href: "#foo" }} />
  {/* 😕 what this will render at all? */}
  <Link href="#bar" root={{ children: () => <Foo /> }}>
    Hello world
  </Link>
</>
```

---

It's one of reasons why I would prefer to use `wrapper` than `root` as a slot:

- due a reason above IMO it's not good idea to expose `root` slot
  - then it cannot be consistent across components: some components will have `root` in `props`, some will not, but all have `root` slot
- it _should not be_ (???) consistent (as components like `Input` are exceptions)